### PR TITLE
Implemented lpvip masked and non-masked softmax

### PR DESCRIPTION
**Pull request by abracodeabra1**
# xlnscpp PR: Add LPVIP32 softmax (scale + masked)

## Summary

Add `xlns16_softmax_lpvip32` and `xlns16_softmax_masked_lpvip32` in `xlns32lpvip.cpp`, aligned with `xlns16_softmax` / `xlns16_softmax_masked`:

```
softmax_lpvip32(a, scale)        = exp(scale*a[i] - max) / sum_j exp(scale*a[j] - max)
softmax_masked_lpvip32(a, mask)  = same, after applying mask to scaled logits
```

Mask semantics match xlns16.

Only the normalization sum uses LPVIP (`xlns16_sum_lpvip32`). Scale, mask bias, and max-subtraction use plain `xlns16_mul` / `xlns16_add` / `xlns16_sub` / `xlns16_exp` / `xlns16_div` — same as xlns16 — so short-vector accuracy stays close to the baseline while long sums still benefit from 32-bit LPVIP accumulation.

Also fixes `xlns16_sum_lpvip32` to initialize the accumulator as `((xlns32)a[0])<<16` (matching `vec_dot_lpvip32`); the previous unshifted start corrupted sums when element 0 dominated.

## Motivation

`summation16bit.MD` and `test16lpvip32monte.cpp` already reference `xlns16_softmax_lpvip32`, but it was unimplemented. Softmax on long vectors benefits from LPVIP summation the same way sum / dot / layernorm do. Routing max-sub through LPVIP was tried and hurt short-vector accuracy; keeping plain xlns16 for those ops matches the original “LPVIP for summation” intent.

## Testing

Compared vs float reference:

Ideal (`xlns16_ideal`):

```
plain softmax max abs diff vs float ref: 0.004406
dense grid (n_samples=14406, vector size 6):
  max abs 0.013552  max rel (ref>0.05) 2.8915%
scaled softmax (1/sqrt(64)) max abs diff: 0.002968
masked softmax max abs diff: 0.002802
```

Table + alt (`xlns16_table` + `xlns16_alt`): similar margins (dense grid vs float abs 0.012319 / rel 3.0010%).

## Notes

- Default `scale = xlns16_one` keeps existing 3-arg call sites (e.g. `test16lpvip32monte.cpp`) compiling unchanged.

### CODE REVIEW

**Comment by markgarnold** [comment]
Proper to limit lpvip32 to summation
